test/test_cost.py

```diff
@@ -577,37 +603,38 @@ def test_distributional_dqn(
             p.data += torch.randn_like(p)
         assert all((p1 != p2).all() for p1, p2 in zip(parameters, actor.parameters()))
 
-    def test_dqn_notensordict(self):
+    @pytest.mark.parametrize("observation_key", ["observation", "observation2"])
+    @pytest.mark.parametrize("reward_key", ["reward", "reward2"])
+    @pytest.mark.parametrize("done_key", ["done", "done2"])
```

Comment:
do we skip the action key by design?

Reply:
No it is intended.
Atm, the action key cannot really be configured since it is used in the constructor of the DQN loss via `_find_action_space(action_space)`. 

Either I need to remove the action key from the configurable keys or the following part of the constructor must be moved until `.set_keys()` is called:
```
        if action_space is None:
            # infer from value net
            try:
                action_space = value_network.spec
            except AttributeError:
                # let's try with action_space then
                try:
                    action_space = value_network.action_space
                except AttributeError:
                    raise ValueError(self.ACTION_SPEC_ERROR)
        if action_space is None:
            warnings.warn(
                "action_space was not specified. DQNLoss will default to 'one-hot'."
                "This behaviour will be deprecated soon and a space will have to be passed."
                "Check the DQNLoss documentation to see how to pass the action space. "
            )
            action_space = "one-hot"
        self.action_space = _find_action_space(action_space)
```

Reply:
Got it thanks!